expo/expo
# [fingerprint] Add more debug info

# Why

One thing I've noticed while building the fingerprint runtime policy for expo-updates is that I'm often finding myself comparing two `sources` debug info arrays to see why a hash differs. This is fine up until a certain point, but for `dir` hashes (where all the files within a directory are hashed), there usually isn't enough granularity to determine why they differ.

Closes ENG-11577.

# How

This PR adds an optional verbose `debug` option to fingerprint which, when enabled, prints out the entire tree of hashes. This should make it faster to compare two `sources` down to a precise per-file `dir` hash.

# Test Plan

- Run tests.
- Run `npx expo-updates runtimeversion:resolve --platform android`, see no debug output
- Run `npx expo-updates runtimeversion:resolve --platform android --debug`, see massive debug output.

# Checklist
